### PR TITLE
Rename `ast::Lit` as `ast::MetaItemLit`.

### PR DESCRIPTION
**Pull request by nnethercote**
And some other literal cleanups.

r? @petrochenkov

### CODE REVIEW

File: compiler/rustc_ast/src/ast.rs [hunk 1]
```diff
@@ -1709,20 +1709,19 @@ pub struct MetaItemLit {
     pub span: Span,
 }
 
-/// Same as `Lit`, but restricted to string literals.
+/// A string literal representing an ABI value.
```
Comment (petrochenkov) [inline compiler/rustc_ast/src/ast.rs:1712]: I suspect that `StrLit` will be used in other places too when https://github.com/rust-lang/rust/issues/95625 is fixed, so I'd keep it as is.

Comment (nnethercote) [reply compiler/rustc_ast/src/ast.rs]: Ok, I have replaced that commit with one that only adjust the comments on `StrLit`.

**Comment by matthiaskrgr** [comment]
@bors r- failed in a rollup
https://github.com/rust-lang/rust/pull/104869#issuecomment-1327223630

**Comment by nnethercote** [comment]
I fixed the failure.

@bors r=petrochenkov

**Comment by nnethercote** [comment]
And I fixed the conflict.

@bors r=petrochenkov

**Pull request merged**